### PR TITLE
Migrate to Nextclade v3

### PR DESCRIPTION
**Pull request by joverlee521**
## Description of proposed changes

Migrates the ingest workflow to Nextclade v3, which simplifies the workflow since we no longer need to use two Nextclade datasets. Includes additional clean ups to use `augur curate rename` and `augur merge`, see commits for details.

## Related issue(s)

Resolves https://github.com/nextstrain/mpox/issues/280

## Checklist

- [x] Checks pass
- [x] Trial run of [ingest workflow](https://github.com/nextstrain/mpox/actions/runs/11412617923)

### CODE REVIEW

**Comment by joverlee521** [comment]
Trial run successfully completed and uploaded results to `s3://nextstrain-data/files/workflows/mpox/branch/ingest-nextclade-v3/`

There are many changes across the metadata.tsv compared to the production metadata.tsv because of changes in the Nextclade columns. A majority of these changes come from the `divergence` column, which is expected because the v3 MPXV Nextclade dataset uses NC_063383.1 as the reference instead of a reconstructed ancestor.

I took a quick look at clade changes:
- A majority of clade changes are a result of the new Ia & Ib clade distinctions
- 42 sequences that used to fail now get assigned clades
- 12 sequences that were labelled as `outgroup` now fail 

<details>

<summary> Detailed clade change counts </summary>

|prod\_clade  |v3\_clade    |count  |
|-------------|-------------|------:|
|             |I            |6      |
|             |II           |4      |
|             |IIa          |5      |
|             |IIb          |10     |
|             |Ia           |2      |
|             |outgroup     |15     |
|I            |II           |1      |
|I            |IIb          |1      |
|I            |Ia           |518    |
|I            |Ib           |133    |
|I            |outgroup     |22     |
|II           |IIa          |1      |
|II           |IIb          |14     |
|II           |outgroup     |7      |
|IIa          |II           |2      |
|IIa          |IIb          |24     |
|IIa          |outgroup     |1      |
|IIb          |outgroup     |2      |
|outgroup     |             |12     |
|outgroup     |I            |1      |
|outgroup     |II           |4      |
|outgroup     |IIa          |4      |
|outgroup     |IIb          |28     |

</details>

These changes seem reasonable to me, but would like a quick review from @corneliusroemer

File: ingest/rules/nextclade.smk [hunk 1]
```diff
@@ -1,86 +1,99 @@
...
-        nextclade2 dataset get --name hMPXV --output-zip {output}
+        r"""
+        nextclade3 dataset get \
+            --name {params.dataset_name:q} \
+            --output-zip {output:q}
         """
 
 
-rule align:
+rule run_nextclade:
```
Comment (genehack) [inline ingest/rules/nextclade.smk:17]: log/benchmark?

**Review by genehack — APPROVED** [review]

File: ingest/rules/nextclade.smk (hunk 1 above)
Comment (joverlee521) [reply ingest/rules/nextclade.smk:21]: Added in https://github.com/nextstrain/mpox/pull/281/commits/42c10401366712a3894c067656b3b46b33c8a7fa

File: ingest/rules/nextclade.smk [hunk 2]
```diff
@@ -30,7 +30,7 @@ rule run_nextclade:
         # The lambda is used to deactivate automatic wildcard expansion.
         # https://github.com/snakemake/snakemake/blob/384d0066c512b0429719085f2cf886fdb97fd80a/snakemake/rules.py#L997-L1000
         translations=lambda w: "results/translations/{cds}.fasta",
-    threads: 4
+    threads: workflow.cores
```
Comment (joverlee521) [inline ingest/rules/nextclade.smk:33]: This shaved off 8 mins for the Nextclade job, but only 3 mins off the total workflow. 

Using _all_ of the cores prevents Snakemake from scheduling other concurrent jobs such as `upload_to_s3` jobs. The upload jobs can be slow as well, so we could adjust the threads to something like `workflow.cores * 0.75`. 

It's only a couple minute difference so I'm not going to worry about it for now.

**Review by genehack — APPROVED** [review]

**Pull request merged**